### PR TITLE
add listeners for pingSent & pongReceived 

### PR DESCRIPTION
**Pull request by faizan-carameltech**
I needed to make some decisions on basis of number of failed pings and pong timeouts. Therefore publicly handling listeners for these events.

### CODE REVIEW

File: Source/SocketIOClient.swift [hunk 1]
```diff
@@ -276,6 +276,19 @@ public final class SocketIOClient : NSObject, SocketEngineClient, SocketParsable
         }
     }
 
+    ///ping & pong
+    public func engineDidSendPing() {
+        DefaultSocketLogger.Logger.log("%@", type: logType, args: "")
```
Comment (nuclearace) [inline Source/SocketIOClient.swift]: Instead of an empty string, can you log something like "Sent Ping"?

File: Source/SocketIOClient.swift [hunk 2]
```diff
@@ -276,6 +276,19 @@ public final class SocketIOClient : NSObject, SocketEngineClient, SocketParsable
...
+        
+        handleEvent("sentPing", data: [""], isInternalMessage: true)
```
Comment (nuclearace) [inline Source/SocketIOClient.swift]: The data array can be empty, empty string doesn't mean anything.

File: Source/SocketIOClient.swift [hunk 3]
```diff
@@ -276,6 +276,19 @@ public final class SocketIOClient : NSObject, SocketEngineClient, SocketParsable
...
+    }
+    
+    public func engineDidReceivePong() {
+        DefaultSocketLogger.Logger.log("%@", type: logType, args: "")
+        
+        handleEvent("receivedPong", data: [""], isInternalMessage: true)
```
Comment (nuclearace) [inline Source/SocketIOClient.swift]: The data array can be empty, empty string doesn't mean anything.

File: Source/SocketIOClient.swift [hunk 4]
```diff
@@ -276,6 +276,19 @@ public final class SocketIOClient : NSObject, SocketEngineClient, SocketParsable
...
 
+    ///ping & pong
+    public func engineDidSendPing() {
+        DefaultSocketLogger.Logger.log("%@", type: logType, args: "")
+        
+        handleEvent("sentPing", data: [""], isInternalMessage: true)
+    }
+    
+    public func engineDidReceivePong() {
+        DefaultSocketLogger.Logger.log("%@", type: logType, args: "")
```
Comment (nuclearace) [inline Source/SocketIOClient.swift]: Instead of an empty string, can you log something like "Received Pong"?

**Comment by Dwar3xwar** [comment]
Any update on this?

File: Source/SocketIOClient.swift (hunk 2 above)
Comment (zhangtaotao) [inline Source/SocketIOClient.swift]: if I want to pong time but  The data array can is empty,

**Comment by faizan-carameltech** [comment]
Sorry I was busy with work that I could not keep up with this PR.  @nuclearace are you planing to support this on your own?

**Comment by nuclearace** [comment]
@faizan-carameltech I could add this on my own if you don't think you'd be able to update the PR.

**Comment by faizan-carameltech** [comment]
@nuclearace that would be really great. I was just going through the latest changes you have made in master. It's a lot to keep up with. I have my own deadlines pending and therefore I don't think I will be able to mange this PR in the near future.

I really appreciate the great work you are doing. And I really hope you can put pingSent and pongReceived listener support in your upcoming commits.

**Comment by nuclearace** [comment]
https://github.com/socketio/socket.io-client-swift/commit/c03fc96f04972856c430be9d5f8fc16b376c6875

**Pull request closed without merging**